src/openlcb/BroadcastTimeServer.cxxtest

```diff
@@ -81,6 +81,215 @@ TEST_F(BroadcastTimeServerTest, Create)
     EXPECT_EQ(server_->day_of_year(), 0);
 };
 
+TEST_F(BroadcastTimeServerTest, Time)
+{
+    expect_any_packet();
+    EXPECT_CALL(*this, update_callback()).Times(AtLeast(0));
+
+    // positive rate
+    server_->set_rate_quarters(2000);
+    server_->set_time(0, 0);
+    server_->set_date(1, 1);
+    server_->set_year(1970);
+    wait_for_event_thread();
+
+    EXPECT_EQ(server_->time(), 0);
+    server_->start();
+    usleep(123456);
+    EXPECT_EQ(server_->time(), 62);
+
+    // negative rate
+    server_->stop();
+    server_->set_rate_quarters(-2000);
+    server_->set_time(0, 0);
+    server_->set_date(1, 1);
+    server_->set_year(1970);
+    wait_for_event_thread();
+
+    EXPECT_EQ(server_->time(), 59);
+    server_->start();
+    usleep(123456);
+    EXPECT_EQ(server_->time(), -3);
+}
+
+TEST_F(BroadcastTimeServerTest, FastSecToRealNsecPeriod)
+{
+    long long real_nsec;
+    bool result;
+
+    result = server_->fast_sec_to_real_nsec_period(2000, 60, &real_nsec);
+    EXPECT_TRUE(result);
+    EXPECT_EQ(real_nsec, 120000000LL);
```

Comment:
consider putting the order of arguments as
`EXPECT_EQ(expected, actual);`
This is important because if there is a failure, the error text printed will have these labels.